src/gpio.rs

```diff
@@ -18,6 +21,8 @@ pub trait GpioExt {
 }
 
 /// All unlocked pin modes implement this
+// AJM - Maybe explain what locked/unlocked is in context? I'm not familiar with this chipset,
+// so I'm guessing there is some core functionality this represents?
```

Comment:
I'll add words here. Yes, to avoid 'accidents' it's impossible to change the pin mode for a handful of pins that do important things; impossible unless you do a magic register dance first. Protecting the JTAG pins in this way I understand, but why they chose to lock the two possible NMI pins I don't know. Especially as they then put a button on one of those pins on the developer kit.